web/src/unread_ui.ts

```diff
@@ -30,6 +30,13 @@ export function update_unread_banner(): void {
 
     const filter = narrow_state.filter();
     const is_conversation_view = filter === undefined ? false : filter.is_conversation_view();
+
+    // This place holder is essential since hiding the unread banner would reduce
+    // scrollable place, shifting the message list downward if the scrollbar is
+    // at bottom.
+    $("#mark_read_on_scroll_state_banner_place_holder").html(render_mark_as_read_disabled_banner());
```

Comment:
I like the idea of directly using `render_mark_as_read_disabled_banner()` here but we need to do it for other cases too since the text is different in them and hence the height.

So, you can extract a function which sets the html for both the place holder and visible banner.

Reply:
I updated to use the existing if-else conditions used to determine innerhtml for `#mark_read_on_scroll_state_banner`.